Call to Orchard's `resolve-class` can fail

I hit an exception while doing Clojure stuff.

When Haystack, consequently, did its thing, the following stacktrace was printed:

```
java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: Could not initialize class net.sf.uadetector.service.UADetectorServiceFactory
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at clojure.core$deref_future.invokeStatic(core.clj:2317)
	at clojure.core$future_call$reify__8544.deref(core.clj:7041)
	at clojure.core$deref.invokeStatic(core.clj:2337)
	at clojure.core$pmap$step__8557$fn__8561.invoke(core.clj:7092)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.RT.seq(RT.java:535)
	at clojure.core$seq__5467.invokeStatic(core.clj:139)
	at clojure.core$map$fn__5939.invoke(core.clj:2774)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.Cons.next(Cons.java:39)
	at clojure.lang.RT.next(RT.java:713)
	at clojure.core$next__5451.invokeStatic(core.clj:64)
	at clojure.core.protocols$fn__8249.invokeStatic(protocols.clj:169)
	at clojure.core.protocols$fn__8249.invoke(protocols.clj:124)
	at clojure.core.protocols$fn__8204$G__8199__8213.invoke(protocols.clj:19)
	at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
	at clojure.core.protocols$fn__8236.invokeStatic(protocols.clj:75)
	at clojure.core.protocols$fn__8236.invoke(protocols.clj:75)
	at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
	at clojure.core$reduce.invokeStatic(core.clj:6886)
	at clojure.core$into.invokeStatic(core.clj:6958)
	at clojure.core$mapv.invokeStatic(core.clj:6970)
	at clojure.core$mapv.invoke(core.clj:6970)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$flag_duplicates.invokeStatic(analyzer.clj:213)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$flag_duplicates.invoke(analyzer.clj:202)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze_stacktrace_data.invokeStatic(analyzer.clj:320)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze_stacktrace_data.invoke(analyzer.clj:314)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze_cause.invokeStatic(analyzer.clj:331)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze_cause.invoke(analyzer.clj:323)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze_causes$fn__8561.invoke(analyzer.clj:355)
	at clojure.core$map$fn__5931$fn__5932.invoke(core.clj:2759)
	at clojure.lang.PersistentVector.reduce(PersistentVector.java:343)
	at clojure.core$transduce.invokeStatic(core.clj:6946)
	at clojure.core$into.invokeStatic(core.clj:6962)
	at clojure.core$into.invoke(core.clj:6950)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze_causes.invokeStatic(analyzer.clj:355)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze_causes.invoke(analyzer.clj:351)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze.invokeStatic(analyzer.clj:373)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze.invoke(analyzer.clj:361)
	at cider.nrepl.middleware.stacktrace$analyze_last_stacktrace.invokeStatic(stacktrace.clj:34)
	at cider.nrepl.middleware.stacktrace$analyze_last_stacktrace.invoke(stacktrace.clj:31)
	at cider.nrepl.middleware.stacktrace$handle_analyze_last_stacktrace_op.invokeStatic(stacktrace.clj:40)
	at cider.nrepl.middleware.stacktrace$handle_analyze_last_stacktrace_op.invoke(stacktrace.clj:36)
	at cider.nrepl.middleware.stacktrace$handle_stacktrace.invokeStatic(stacktrace.clj:73)
	at cider.nrepl.middleware.stacktrace$handle_stacktrace.invoke(stacktrace.clj:69)
	at clojure.lang.Var.invoke(Var.java:388)
	at cider.nrepl$wrap_stacktrace$fn__3415.invoke(nrepl.clj:562)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_ns$fn__3367.invoke(nrepl.clj:430)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_complete$fn__3299.invoke(nrepl.clj:143)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at nrepl.middleware.interruptible_eval$interruptible_eval$fn__1329.invoke(interruptible_eval.clj:154)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_version$fn__3457.invoke(nrepl.clj:682)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at nrepl.middleware.load_file$wrap_load_file$fn__1871.invoke(load_file.clj:81)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_content_type$fn__3267.invoke(nrepl.clj:107)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_slurp$fn__3275.invoke(nrepl.clj:121)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_log$fn__3351.invoke(nrepl.clj:287)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_inspect$fn__3343.invoke(nrepl.clj:227)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at nrepl.middleware.caught$wrap_caught$fn__1262.invoke(caught.clj:97)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at nrepl.middleware.lookup$wrap_lookup$fn__1915.invoke(lookup.clj:52)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at nrepl.middleware.completion$wrap_completion$fn__1839.invoke(completion.clj:58)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at refactor_nrepl.middleware$wrap_refactor$fn__3127.invoke(middleware.clj:244)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at nrepl.middleware.session$add_stdin$fn__1447.invoke(session.clj:379)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_trace$fn__3431.invoke(nrepl.clj:642)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_info$fn__3333.invoke(nrepl.clj:208)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_macroexpand$fn__3359.invoke(nrepl.clj:417)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_undef$fn__3447.invoke(nrepl.clj:671)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_apropos$fn__3283.invoke(nrepl.clj:129)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_profile$fn__3383.invoke(nrepl.clj:473)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_spec$fn__3407.invoke(nrepl.clj:546)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_debug$fn__3309.invoke(nrepl.clj:163)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_enlighten$fn__3317.invoke(nrepl.clj:190)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_refresh$fn__3391.invoke(nrepl.clj:510)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_tracker$fn__3439.invoke(nrepl.clj:660)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_test$fn__3423.invoke(nrepl.clj:618)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at nrepl.middleware.sideloader$wrap_sideloader$fn__1952.invoke(sideloader.clj:108)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_format$fn__3325.invoke(nrepl.clj:194)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_resource$fn__3399.invoke(nrepl.clj:536)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_out$fn__3375.invoke(nrepl.clj:464)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_xref$fn__3465.invoke(nrepl.clj:692)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at nrepl.middleware.dynamic_loader$wrap_dynamic_loader$fn__1515.invoke(dynamic_loader.clj:98)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_classpath$fn__3291.invoke(nrepl.clj:137)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at nrepl.middleware.print$wrap_print$fn__1229.invoke(print.clj:234)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at cider.nrepl$wrap_clojuredocs$fn__3473.invoke(nrepl.clj:707)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at nrepl.middleware.session$session$fn__1432.invoke(session.clj:325)
	at nrepl.middleware$wrap_conj_descriptor$fn__993.invoke(middleware.clj:16)
	at nrepl.server$default_handler$fn__2000.invoke(server.clj:141)
	at nrepl.server$handle_STAR_.invokeStatic(server.clj:24)
	at nrepl.server$handle_STAR_.invoke(server.clj:21)
	at nrepl.server$handle$fn__1968.invoke(server.clj:41)
	at clojure.core$binding_conveyor_fn$fn__5823.invoke(core.clj:2047)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class net.sf.uadetector.service.UADetectorServiceFactory
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:466)
	at clojure.lang.RT.classForName(RT.java:2209)
	at clojure.lang.RT.classForName(RT.java:2218)
	at clojure.lang.Compiler.maybeResolveIn(Compiler.java:7455)
	at clojure.core$ns_resolve.invokeStatic(core.clj:4371)
	at clojure.core$ns_resolve.invokeStatic(core.clj:4360)
	at clojure.core$ns_resolve.invoke(core.clj:4360)
	at cider.nrepl.inlined.deps.orchard.v0v14v2.orchard.java$resolve_class$fn__8313.invoke(java.clj:301)
	at cider.nrepl.inlined.deps.orchard.v0v14v2.orchard.java$resolve_class.invokeStatic(java.clj:301)
	at cider.nrepl.inlined.deps.orchard.v0v14v2.orchard.java$resolve_class.invoke(java.clj:295)
	at cider.nrepl.inlined.deps.orchard.v0v14v2.orchard.java$resolve_symbol.invokeStatic(java.clj:335)
	at cider.nrepl.inlined.deps.orchard.v0v14v2.orchard.java$resolve_symbol.invoke(java.clj:321)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze_fn.invokeStatic(analyzer.clj:91)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze_fn.invoke(analyzer.clj:70)
	at clojure.core$comp$fn__5876.invoke(core.clj:2586)
	at clojure.core$comp$fn__5876.invoke(core.clj:2586)
	at clojure.core$comp$fn__5876.invoke(core.clj:2586)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze_frame.invokeStatic(analyzer.clj:220)
	at cider.nrepl.inlined.deps.haystack.v0v0v3.haystack.analyzer$analyze_frame.invoke(analyzer.clj:215)
	at clojure.core$partial$fn__5908.invoke(core.clj:2641)
```

Perhaps a simple try/catch might suffice?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,12 @@
 
 ## master (unreleased)
 
+### Bugs fixed
+
+* [#9](https://github.com/clojure-emacs/haystack/issues/9): handle unloadable classes.
+
+## 0.0.3
+
 ### Changes
 
 * Add stacktrace analyzer and parsers.

--- a/src/haystack/analyzer.clj
+++ b/src/haystack/analyzer.clj
@@ -87,11 +87,16 @@
                                    path->url
                                    str)
                            (str (frame->url frame)))))
-    (assoc frame :file-url (some->> frame :name symbol
-                                    (java/resolve-symbol 'user)
-                                    :file
-                                    path->url
-                                    str))))
+    (assoc frame :file-url (try
+                             (some->> frame :name symbol
+                                      (java/resolve-symbol 'user)
+                                      :file
+                                      path->url
+                                      str)
+                             (catch Throwable _
+                               ;; `java/resolve-symbol` can throw exceptions when the underlying class cannot be loaded.
+                               ;; See https://github.com/clojure-emacs/haystack/issues/9
+                               nil)))))
 
 (defn- analyze-file
   "Associate the file type (extension) of the source file to the frame map, and

